community/kernel/src/main/java/org/neo4j/kernel/NeoStoreDataSource.java

```diff
@@ -228,6 +235,8 @@ boolean applicable( DiagnosticsPhase phase )
     }
 
     public static final String DEFAULT_DATA_SOURCE_NAME = "nioneodb";
+    private final boolean failOnCorruptedLogFiles = FeatureToggles.flag( NeoStoreDataSource.class,
+            "failOnCorruptedLogFiles", false );
```

Comment:
A default value of `false` is potentially unsafe isn't it? I see this recovery feature as something like:

- You do recovery and for some reason it fails
- You realize that there may be some corruption and perhaps do some preparation work, backup, copy whatever
- You do recovery with this flag to force attempt recovery anyway

Reply:
i do not think we should ask user confirmation for recovery it should just work, also consider scenarios like cloud, docker containers auto restarts, copying of tx logs over network etc - that should not involve any user interactions at all. And since we have a first safety mechanic to backup files that will be truncated and the fact that we doing that in brand new version and will have time to catch possible problems there i think it should be "false" by default